### DISCUSSION TITLE
API request question

### DISCUSSION THREAD

**Discussion by ilovemypinktutu** [osm-search/Nominatim#3568] [Q&A]
Hello,

I need to convert about 3,000 addresses into lat and lon. I was testing API search request using python for maybe 3 to 5 times manually. Soon, I got a message that my access is forbidden with response 403. I read the usage policy and I don't think I violated anything -- I didn't automate anything, all my requests were at least a couple of minutes apart, and the total number of queries doesn't exceed 5. Can someone correct me what I'm doing wrong and how to gain access again?

Here's what the request I made in python.

url = 'https://nominatim.openstreetmap.org/search?q=1626+Oak+Street+La+Crosse,+WI+54602'
r = re.get(url)

Thanks a lot!
Bo

**Comment by mtmail** [comment]
Set a user agent identifying your application "stock User-Agents as set by http libraries will not do" https://operations.osmfoundation.org/policies/nominatim/
